karate-core/src/main/java/com/intuit/karate/driver/chrome/Chrome.java

```diff
@@ -57,13 +57,20 @@ public static Chrome start(ScenarioContext context, Map<String, Object> map, Log
                 FileUtils.isOsWindows() ? DEFAULT_PATH_WIN : FileUtils.isOsMacOsX() ? DEFAULT_PATH_MAC : DEFAULT_PATH_LINUX);
         options.arg("--remote-debugging-port=" + options.port);
         options.arg("--no-first-run");
-        if (options.userDataDir != null) {
-            options.arg("--user-data-dir=" + options.userDataDir);
-        }
         options.arg("--disable-popup-blocking");
         if (options.headless) {
             options.arg("--headless");
         }
+        if (options.userDataDir != null) {
+            options.arg("--user-data-dir=" + options.userDataDir);
+        }
+        //Chrome Custom Stuff
+        if (options.userAgent != null) {
```

Comment:
if you feel `disableNotifications` is a "must have" for Chrome you can make it that way like `no-first-run`.

note that anything else, such as `user-agent` can be done via `addOptions`

one of the principles is that the options should be "browser neutral" as far as possible. although `headless` is a special case I admit

would you be able to update the `karate-core/README.md` also to reflect the changes in both PR-s

thanks for contributing and your feedback

Reply:
Alright, you can leave this like that.
The problem was solved in the DriverOptions.java file.